pkg/controller/containerjfr/containerjfr_controller.go

```diff
@@ -0,0 +1,276 @@
+		err = r.client.Create(context.TODO(), pod)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		exporter := newExporterServiceForPod(pod.ObjectMeta.Namespace)
+		err = r.client.Create(context.TODO(), exporter)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		cmdChan := newCommandChannelServiceForPod(pod.ObjectMeta.Namespace)
+		err = r.client.Create(context.TODO(), cmdChan)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// Pod created successfully - don't requeue
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Pod already exists - don't requeue
+	reqLogger.Info("Skip reconcile: Pod already exists", "Pod.Namespace", found.Namespace, "Pod.Name", found.Name)
+	return reconcile.Result{}, nil
+}
+
+func newPodForCR(cr *rhjmcv1alpha1.ContainerJFR) *corev1.Pod {
+	labels := map[string]string{
+		"app": cr.Name,
+	}
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-pod",
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  cr.Name,
+					Image: "quay.io/rh-jmc-team/container-jfr:0.4.0-debug",
+					Ports: []corev1.ContainerPort{
+						{
+							ContainerPort: 8181,
+						},
+						{
+							ContainerPort: 9090,
+						},
+						{
+							ContainerPort: 9091,
+						},
+					},
+					Env: []corev1.EnvVar{
+						{
+							Name:  "CONTAINER_JFR_WEB_PORT",
+							Value: "8181",
+						},
+						{
+							Name:  "CONTAINER_JFR_EXT_WEB_PORT",
+							Value: "80",
+						},
+						{
+							Name: "CONTAINER_JFR_WEB_HOST",
+							ValueFrom: &corev1.EnvVarSource{
+								ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "containerjfr",
+									},
+									Key: "CONTAINER_JFR_WEB_HOST",
+								},
+							},
+						},
+						{
+							Name:  "CONTAINER_JFR_LISTEN_PORT",
+							Value: "9090",
+						},
+						{
+							Name:  "CONTAINER_JFR_EXT_LISTEN_PORT",
+							Value: "80",
+						},
+						{
+							Name: "CONTAINER_JFR_LISTEN_HOST",
+							ValueFrom: &corev1.EnvVarSource{
+								ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "containerjfr-command",
+									},
+									Key: "CONTAINER_JFR_LISTEN_HOST",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func newExporterServiceForPod(namespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "containerjfr",
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app": "containerjfr",
+			},
+			Annotations: map[string]string{
+				"fabric8.io/expose": "true",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Selector: map[string]string{
+				"app": "containerjfr",
```

Comment:
Perhaps to make this more flexible, we should use the name of the ContainerJFR object, like the pod does for its app label.

Reply:
Done.